BlueprintUICommonControls/Sources/ScrollView.swift

```diff
@@ -267,3 +273,23 @@ fileprivate final class ScrollerWrapperView: UIView {
     }
 
 }
+
+private final class RefreshControl: UIRefreshControl {
+
+    override var frame: CGRect {
+        get {
+            return super.frame
+        }
+        set {
+            // UIRefreshControl frame does not account for UIScrollView's left inset. Offset the
+            // frame with negative inset so that the `refreshControl` is centered with respect to the content.
+            var newFrame = newValue
+            if let parentScrollView = superview as? UIScrollView {
```

Comment:
This is very surprising behavior for a property. Even if it's a private implementation detail, it's pretty fragile – it makes assumptions about the view hierarchy, and it assumes that it is the child view of a scroll view at the time frame is assigned. It's also surprising that the parent can assign one frame, but a different frame is actually assigned... This effectively means that the child is making layout decisions about itself. Parents should always be entirely responsible for the layout of their child views, and this breaks that contract. The following code seems to work well in a sample project. It accomplishes the same thing, but it keep layout decisions in the parent view:

```
final class CustomScrollView: UIScrollView {
    
    override func layoutSubviews() {
        super.layoutSubviews()
        refreshControl?.center.x = bounds.midX
    }
    
}
```

Reply:
(Also, does this happen in a plain UIScrollView in a new project?)

Reply:
It does, I just tested.

Reply:
Yay...

Reply:
Thanks, will update. Originally had the refresh control layout in `layoutSubviews` but for some reason it wasn't laying out correctly.